pkg/common/cns-lib/volume/manager.go

```diff
@@ -210,20 +220,82 @@ func (m *defaultManager) CreateVolume(ctx context.Context, spec *cnstypes.CnsVol
 			spec.Metadata.ContainerCluster.VSphereUser = s.UserName
 		}
 
-		// Construct the CNS VolumeCreateSpec list.
-		var cnsCreateSpecList []cnstypes.CnsVolumeCreateSpec
-		var task *object.Task
-		var taskInfo *vim25types.TaskInfo
-		// Store the volume name passed in by input spec, this name may exceed 80 characters.
-		volNameFromInputSpec := spec.Name
-		// Call the CNS CreateVolume.
-		taskDetailsInMap, ok := volumeTaskMap[volNameFromInputSpec]
-		if ok {
-			task = taskDetailsInMap.task
-			log.Infof("CreateVolume task still pending for Volume: %q, with taskInfo: %+v",
-				volNameFromInputSpec, task)
+		var (
```

Comment:
Can we break down the CreateVolume method? Currently it is 300 lines of code!

Reply:
Done!